core/examples/luigi-core-material-ui/src/App.js

```diff
@@ -0,0 +1,253 @@
+        [theme.breakpoints.up('md')]: {
+          padding: 0,
+          minWidth: 0
+        }
+      }
+    },
+    MuiIconButton: {
+      root: {
+        padding: theme.spacing(1)
+      }
+    },
+    MuiTooltip: {
+      tooltip: {
+        borderRadius: 4
+      }
+    },
+    MuiDivider: {
+      root: {
+        backgroundColor: '#404854'
+      }
+    },
+    MuiListItemText: {
+      primary: {
+        fontWeight: theme.typography.fontWeightMedium
+      }
+    },
+    MuiListItemIcon: {
+      root: {
+        color: 'inherit',
+        marginRight: 0,
+        '& svg': {
+          fontSize: 20
+        }
+      }
+    },
+    MuiAvatar: {
+      root: {
+        width: 32,
+        height: 32
+      }
+    }
+  }
+};
+
+const drawerWidth = 256;
+
+const styles = {
+  root: {
+    display: 'flex',
+    minHeight: '100vh'
+  },
+  drawer: {
+    [theme.breakpoints.up('sm')]: {
+      width: drawerWidth,
+      flexShrink: 0
+    }
+  },
+  app: {
+    flex: 1,
+    display: 'flex',
+    flexDirection: 'column'
+  },
+  main: {
+    flex: 1,
+    padding: theme.spacing(6, 4),
+    background: '#eaeff1'
+  },
+  footer: {
+    padding: theme.spacing(2),
+    background: '#eaeff1'
+  }
+};
+
+const listeners = [];
+const removeListeners = () =>
+  listeners.forEach(id => window.Luigi.navigation().removeEventListener(id));
+function App(props) {
+  const { classes } = props;
+  const [mobileOpen, setMobileOpen] = React.useState(false);
+  const [leftNav, setLeftNav] = React.useState({ categories: [] });
+  const [topNav, setTopNav] = React.useState({ topNavData: { children: [] } });
+
+  const handleDrawerToggle = () => {
+    setMobileOpen(!mobileOpen);
+  };
+
+  const prepareCategories = childrenObj => {
+    const categories = [];
+    for (const category in childrenObj) {
+      if (!category.startsWith('__')) {
+        categories.push({
+          label: category,
+          children: childrenObj[category]
+        });
+      }
+    }
+    // categories = [
+    //   {
+    //     id: 'Develop',
+    //     children: [
+    //       { id: 'Authentication', icon: <PeopleIcon />, active: true },
+    return categories;
+  };
+
+  const handleLuigiEvents = (name, data) => {
+    if (!data) {
+      return;
+    }
+
+    if (name.endsWith('topNav')) {
+      setTopNav(data);
+    }
+    if (name.endsWith('leftNav')) {
+      data.categories = prepareCategories(data.leftNavData.children);
+      setLeftNav(data);
+    }
+  };
+
+  useEffect(() => {
+    removeListeners();
+    listeners.push(
+      window.Luigi.navigation().addEventListener('leftNav', data =>
+        handleLuigiEvents('leftNav', data)
+      )
+    );
+    listeners.push(
+      window.Luigi.navigation().addEventListener('topNav', data =>
+        handleLuigiEvents('topNav', data)
+      )
+    );
+    // listeners.push(
+    //   window.Luigi.navigation().addEventListener('tabNav', data =>
+    //     handleLuigiEvents('tabNav', data)
+    //   )
+    // );
```

Comment:
Clean up?

Reply:
I wanted to make it simpler for ppl to set up a tab navigation, this would be a starting point.